fontforge/namelist.c

```diff
@@ -184,7 +184,7 @@ const char *StdGlyphName(char *buffer, int uni,enum uni_interp interp,NameList *
     if ( (uni>=0 && uni<' ') ||
 	    (uni>=0x7f && uni<0xa0) )
 	/* standard controls */;
-    else if ( uni!=-1  ) {
+    else if ( uni>0 && uni <= 0x10ffff ) {
```

Comment:
*Could* preferentially check the NameList if it comes in non-NULL for larger values, but only if we reliably know how long it is.

Reply:
...okay, I took a more careful look here. We have several balls up in the air here, so trying to juggle this is a bit interesting... to begin with, the code comes from a time when ints were maybe 16 bits long. Since this code can end up on commandline with old/ancient PCs, and to play nice with those older PCs, we should probably use long instead of int for these variables, use Unicode/ustring.c:char *utf8_idpb() as a reference. I would suggest changing int up, ub, uc; to long up, ub, uc; that ensures all PCs are using 32bit int, and then the sprintf at 206..208 should be printing long, not int, you can see an example of the 'l' here... libuninameslist/tests/call-test.c:42..44, this also shows we need to consider values larger than ffffh that then require more than 4 hex digits ...there's probably a better example that uses 5,6,7,8 digits elsewhere but I can't recall where ATM.
My error for mentioning 10ffff earlier, the unicode range is 0...11ffffh (looking at makeutype.c:135). Another max point is MAXC seen at line 126, or if you want to use it within FontForge, it is FF_UTYPE_MAXC which is seen at inc/utype.h:43   FF_UTYPE_MAXC is where NameList stops, so everything at this point and higher will only have uni values or hex codes.

Reply:
I take it that if I change the prototype of `uni` to long I don't need to worry about the callers, as an int can always be converted to a long? Or are you suggesting a more systemic change up the calling hierarchy?